apps/emqx_resource/src/emqx_resource_worker.erl

```diff
@@ -674,12 +816,14 @@ inflight_size(Name) ->
 
 inflight_append(undefined, _Ref, _Query, _Id, _Index) ->
     ok;
-inflight_append(Name, Ref, [?QUERY(_, _, _) | _] = Batch, Id, Index) ->
-    ets:insert(Name, {Ref, [?QUERY(From, Req, true) || ?QUERY(From, Req, _) <- Batch]}),
+inflight_append(Name, Ref, [?QUERY(_, _, _) | _] = Batch0, Id, Index) ->
+    Batch = mark_as_sent(Batch0),
+    ets:insert(Name, {Ref, Batch}),
     emqx_resource_metrics:inflight_set(Id, Index, inflight_size(Name)),
```

Comment:
Fixme: if there are batches in the inflight ETS, the inflight metric will track number of batches rather than number of messages.